code/modules/research/xenobiology/xenobio_camera.dm

```diff
@@ -109,72 +122,93 @@
-		connected_recycler.connected += src
-		return TRUE
-
-/*
-Boilerplate check for a valid area to perform a camera action in.
-Checks if the AI eye is on a valid turf and then checks if the target turf is xenobiology compatible
-Due to keyboard shortcuts, the second one is not necessarily the remote eye's location.
-*/
-/obj/machinery/computer/camera_advanced/xenobio/proc/validate_area(mob/living/user, mob/eye/camera/remote/xenobio/remote_eye, turf/open/target_turf)
-	if(!GLOB.cameranet.checkTurfVis(remote_eye.loc))
-		to_chat(user, span_warning("Target is not near a camera. Cannot proceed."))
+	if(.)
+		return .
+
+	if(istype(tool, /obj/item/slimepotion/slime))
+		return slimepotion_act(user, tool)
+	if(istype(tool, /obj/item/food/monkeycube))
+		return monkeycube_act(user, tool)
+	if(istype(tool, /obj/item/storage/bag) || istype(tool, /obj/item/storage/box/monkeycubes))
+		return storage_act(user, tool)
+	return NONE
+
+/// Handles inserting a slime potion into the console, potentially swapping out an existing one.
+/obj/machinery/computer/camera_advanced/xenobio/proc/slimepotion_act(mob/living/user, obj/item/slimepotion/slime/used_potion)
+	if(!user.transferItemToLoc(used_potion, src))
+		balloon_alert(user, "can't insert!")
+		return ITEM_INTERACT_BLOCKING
+
+	if(!QDELETED(current_potion))
+		try_put_in_hand(current_potion, user)
+		balloon_alert(user, "swapped")
+	else
+		balloon_alert(user, "inserted")
+
+	current_potion = used_potion
+	xeno_hud.update_potion(current_potion)
+	return ITEM_INTERACT_SUCCESS
+
+/// Handles inserting a monkey cube into the console.
+/obj/machinery/computer/camera_advanced/xenobio/proc/monkeycube_act(mob/living/user, obj/item/food/monkeycube/used_cube)
+	stored_monkeys += 1
+	balloon_alert(user, "inserted")
+	to_chat(user, span_notice("You feed [used_cube] to [src]. It now has [stored_monkeys] monkey cubes stored."))
+	xeno_hud.on_update_hud(LAZYLEN(stored_slimes), stored_monkeys, max_slimes)
+	qdel(used_cube)
+	return ITEM_INTERACT_SUCCESS
+
+/// Handles inserting any monkey cubes stored in the tool into the console.
+/obj/machinery/computer/camera_advanced/xenobio/proc/storage_act(mob/living/user, obj/item/tool)
+	var/loaded_any = FALSE
+	for(var/obj/storage_item in tool.contents)
+		if(istype(storage_item, /obj/item/food/monkeycube))
+			loaded_any = TRUE
+			stored_monkeys += 1
+			qdel(storage_item)
+	if(!loaded_any)
+		balloon_alert(user, "no monkey cubes!")
+		return ITEM_INTERACT_BLOCKING
+
+	balloon_alert(user, "loaded monkey cubes")
+	to_chat(user, span_notice("You fill [src] with the monkey cubes stored in [tool]. [src] now has [stored_monkeys] monkey cubes stored."))
+	xeno_hud.on_update_hud(LAZYLEN(stored_slimes), stored_monkeys, max_slimes)
+	return ITEM_INTERACT_SUCCESS
+
+/obj/machinery/computer/camera_advanced/xenobio/multitool_act(mob/living/user, obj/item/multitool/tool)
+	if(!istype(tool)) // Needed as long as this uses a var on the multitool.
+		return ITEM_INTERACT_BLOCKING
```

Comment:
`return NONE` is probably preferred for "invalid tool", to prevent accidental blocking.